starknet_devnet/blueprints/feeder_gateway.py

```diff
@@ -210,6 +212,18 @@ async def get_transaction_status():
     tx_status = await state.starknet_wrapper.transactions.get_transaction_status(
         transaction_hash
     )
+
+    # In case of blocks-on-demand feature tx_status should be received
+    if (
```

Comment:
it can be like this and no action is needed I just want to mention that I like this approach with many ifs:
```
is_block_on_demand = state.starknet_wrapper.config.blocks_on_demand
is_transaction_not_recived = tx_status["tx_status"] == TransactionStatus.NOT_RECEIVED.name
is_any_of_transaction = any(hex(transaction.transaction_hash) == transaction_hash for transaction in state.starknet_wrapper.pending_txs)

if is_block_on_demand and is_transaction_not_recived and is_any_of_transaction:
    tx_status["tx_status"] = TransactionStatus.RECEIVED.name
```